todo-app/src/Add-Todo.js

```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import ListItem from "./List-Item";
+
+let todoItems = [];
+
+export default class AddTodo extends React.Component {
+
+    state = {
+        description: "enter todo item",
+        deadline: new Date(),
+        done: false
+    }
+
+    handleTextChange = e => {
+        const description = e.target.value;
+        this.setState({ description });
+    }
+
+    handleDateChange = e => {
+        const deadline = e.target.value;
+        this.setState({ deadline });
+    }
+
+    // render new todo when Add button is clicked
+    handleButtonClick = () => {
+        const { description, deadline } = this.state;
+        todoItems.push({ description, deadline, done: false });
+
+        this.setState({ description, deadline });
```

Comment:
What is this `setState` call? You seem to be fetching the state here, and then setting to state back to what it already was. This is not what we want to do.

In fact, because this is your add-todo function, we want to _reset_ the values here to make the form empty again.

Moreover, in relation to your incorrect state management, this `setState` call is the only reason your app is actually working, as you are storing your `todos` array outside React's state, so it has no idea when this array changes.